clients/drcachesim/common/trace_entry.h

```diff
@@ -747,6 +747,123 @@ struct schedule_entry_t {
     uint64_t instr_count;
 } END_PACKED_STRUCTURE;
 
+#ifdef BUILD_PT_TRACER
+
+/**
+ * The type of a syscall PT entry in the offline output.
+ */
+typedef enum {
+    SYSCALL_PT_ENTRY_TYPE_PID = 0,
+    SYSCALL_PT_ENTRY_TYPE_THREAD,
+    SYSCALL_PT_ENTRY_TYPE_PT_METADATA_BOUNDARY,
+    SYSCALL_PT_ENTRY_TYPE_PT_DATA_BOUNDARY,
+    SYSCALL_PT_ENTRY_TYPE_SYSNUM,
+    SYSCALL_PT_ENTRY_TYPE_SYSCALL_ID,
+    SYSCALL_PT_ENTRY_TYPE_SYSCALL_ARGS_NUM,
+    SYSCALL_PT_ENTRY_TYPE_MAX = 6
+} syscall_pt_entry_type_t;
+
+START_PACKED_STRUCTURE
+struct _syscall_pt_entry_t {
+    union {
+        struct {
+            uint64_t pid : 61;
+            uint64_t type : 3;
+        } pid;
+        struct {
+            uint64_t tid : 61;
+            uint64_t type : 3;
+        } tid;
+        union {
+            struct {
+                uint64_t data_size : 61;
+                uint64_t type : 3;
+            } pt_metadata_boundary;
+            struct {
+                uint64_t data_size : 61;
+                uint64_t type : 3;
+            } pt_data_boundary;
+        };
+        struct {
+            uint64_t sysnum : 61;
+            uint64_t type : 3;
+        } sysnum;
+        struct {
+            uint64_t id : 61;
+            uint64_t type : 3;
+        } syscall_id;
+        struct {
+            uint64_t args_num : 61;
+            uint64_t type : 3;
+        } syscall_args_num;
+        uint64_t combined_value;
+    };
+} END_PACKED_STRUCTURE;
+typedef struct _syscall_pt_entry_t syscall_pt_entry_t;
+
+/**
+ * The per-thread metadata and all syscalls' PT data are stored in one PT DATA
+ * Buffer(PDB). The PDB format is:
+ * +----------+----------------------+
+ * |PDB Header|PT_METADATA or PT_DATA|
+ * +----------+----------------------+
+ *
+ * The PDB Header is a list of syscall_pt_entry_t. There are three types of PDB Header:
+ * a. The format of per-thread metadata's PDB header is:
+ * +---+---+--------------------+
+ * |pid|tid|pt_metadata_boundary|
+ * +---+---+--------------------+
+ *
+ * b. The format of syscalls' PT data's PDB header is:
+ * +---+---+----------------+------+----------+---------------------+---------+
+ * |pid|tid|pt_data_boundary|sysnum|syscall_id|syscall_args_boundary|arguments|
```

Comment:
Do you mean s/syscall_args_boundary/syscall_args_num/?

Reply:
Done.